File: cilicili-cloud/cilicili-upload/src/main/java/com/gaoyuanming/upload/controller/VideoController.java
package com.gaoyuanming.upload.controller;

import com.gaoyuanming.upload.model.entity.Video;
import com.gaoyuanming.upload.model.entity.VideoLike;
import com.gaoyuanming.upload.service.VideoLikeService;
import com.gaoyuanming.upload.service.VideoService;
import com.gaoyuanming.upload.service.impl.VideoLikeServiceImpl;
import org.springframework.web.bind.annotation.*;

import javax.annotation.Resource;
import java.util.HashMap;
import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/video")
public class VideoController {

    @Resource
    private VideoService videoService;

    @Resource
    private VideoLikeService videoLikeService;

    @GetMapping("")
    public List<Video> findAll() {
        return videoService.findAll();
    }

    @PostMapping("/like")
    public void addLike(@RequestBody VideoLike videoLike) throws Exception {
        videoLikeService.save(videoLike);
    }

    @GetMapping("/findByMusicId")
    public List<Video> findByMusicId(String musicId) {
        return videoService.findAllByMusicId(musicId);
    }

    @GetMapping("/findAllByFollow")
    public List<Video> findAllByFollow(String userId) {
        return videoService.findAllByFollow(userId);
    }

    @GetMapping("/findByUserId")
    public List<Video> findByUserId(String userId) {
        return videoService.findAllByUserId(userId);
    }

    @GetMapping("/findByStatus")
    public List<Video> findByStatus(String status) {
        return videoService.findAllByStatus(Boolean.parseBoolean(status));
    }

    @GetMapping("/findUserLikeAndUpload/{id}")
    public Map<String, List<Video>> findUserLikeAndUpload(@PathVariable String id) {
        Map<String, List<Video>> map = new HashMap<>();
        map.put("uploadList", videoService.findAllByUserId(id));
        map.put("likeList", videoService.findByUserLike(id));
        return map;
    }

    @PostMapping("")
    public Video save(@RequestBody Video video) throws Exception {
        video.setStatus(false);
        return videoService.save(video);
    }

    @PutMapping("/{id}/setStatus")
    public void setStatus(@PathVariable String id, String status) {
        System.out.println(status);
        videoService.setStatus(id, Boolean.parseBoolean(status));
    }

}

File: cilicili-go/api/upload/video_controller.go
package upload

import (
	"cilicili-go/model/entity"
	"cilicili-go/model/support"
	"cilicili-go/service/upload"
	"github.com/kataras/iris/v12"
	"strconv"
)

var videoService = new(upload.VideoService)

func saveVideo(c iris.Context) {
	video := new(entity.Video)
	err := c.ReadJSON(video)
	if err != nil {
		_, _ = c.JSON(support.Error(400, err.Error()))
		return
	}
	videoService.Create(video)
	_, _ = c.JSON(support.Success())
}

func setVideoStatus(c iris.Context) {
	id := c.Params().Get("id")
	status := c.URLParam("status")
	b, err := strconv.ParseBool(status)
	if err != nil {
		_, _ = c.JSON(support.Error(400, err.Error()))
		return
	}
	videoService.SetStatus(id, b)
	_, _ = c.JSON(support.Success())
}

func findAllVideo(c iris.Context) {
	_, _ = c.JSON(support.SuccessWithData(videoService.FindAll()))
}

func addVideoLike(c iris.Context) {
	like := new(entity.VideoLike)
	err := c.ReadJSON(like)
	if err != nil {
		_, _ = c.JSON(support.Error(400, err.Error()))
		return
	}
	err = videoService.Like(like)
	if err != nil {
		_, _ = c.JSON(support.Error(400, err.Error()))
		return
	}
	_, _ = c.JSON(support.Success())
}

func findByMusicId(c iris.Context) {
	arr := videoService.FindAllByUserId(c.URLParam("musicId"))
	_, _ = c.JSON(support.SuccessWithData(arr))
}

func findAllByFollow(c iris.Context) {
	param := c.URLParam("userId")
	videos := videoService.FindAllByFollow(param)
	_, _ = c.JSON(support.SuccessWithData(videos))
}

func findByUserId(c iris.Context) {
	_, _ = c.JSON(support.SuccessWithData(videoService.FindAllByUserId(c.URLParam("userId"))))
}

func findByStatus(c iris.Context) {
	status := c.URLParam("status")
	b, err := strconv.ParseBool(status)
	if err != nil {
		_, _ = c.JSON(support.Error(400, err.Error()))
		return
	}
	_, _ = c.JSON(support.SuccessWithData(videoService.FindAllByStatus(b)))
}

func findUserLikeAndUpload(c iris.Context) {
	m := make(map[string][]*entity.Video)
	m["uploadList"] = videoService.FindAllByUserId(c.Params().Get("id"))
	m["likeList"] = videoService.FindByUserLike(c.Params().Get("id"))
	_, _ = c.JSON(support.SuccessWithData(m))
}
